src/appleseed-max-impl/utilities.cpp

```diff
@@ -128,6 +129,29 @@ WStr replace_extension(const WStr& file_path, const WStr& new_ext)
     return new_file_path;
 }
 
+void update_map_buttons(IParamMap2* param_map)
+{
+    if (param_map == nullptr)
+        return;
+
+    IParamBlock2* param_block = param_map->GetParamBlock();
+
+    const int count = param_block->NumParams();
+    for (int i = 0; i < count; ++i)
+    {
+        const int param_id = param_block->IndextoID(i);
+        const ParamDef& param_def = param_block->GetParamDef(param_id);
+        if (param_def.type == TYPE_TEXMAP && (param_def.flags & P_NO_AUTO_LABELS))
+        {
+            const auto texmap = param_block->GetTexmap(param_id, GetCOREInterface()->GetTime(), FOREVER);
+            if (texmap == nullptr)
```

Comment:
That looks perfectly fine. Here's a compact alternative:

    param_map->SetText(param_id, texmap == nullptr ? L"" : L"M");